src/record/record.js

```diff
@@ -151,17 +151,18 @@ Record.prototype.set = function( pathOrData, dataOrCallback, callback ) {
 	var newValue = jsonPath.set( oldValue, path, data, this._options.recordDeepCopy );
 
 	if ( oldValue === newValue ) {
+		if ( typeof callback === 'function' ) utils.setImmediate( callback, null );
```

Comment:
Yup, I think that's what we should do. If you do a write and your offline, you get told immediately it failed because it was offline. Doesn't meant the write was aborted though, just that your offline so you won't be told when its done.